### DISCUSSION TITLE
Unpack error while trying to write into register

### DISCUSSION THREAD

**Discussion by PaulOckenden** [davidrapan/ha-solarman#130] [Q&A]
Anyone know what this error means? 

write_multiple_holding_registers. unpack requires a buffer of 132 bytes

![IMG_1784](https://github.com/user-attachments/assets/087e78e6-88c7-48fb-a51b-afb0c4c37241)

(I’m trying to set one of the parameters used in TOU mode). 

(And probably being an idiot!)

Thanks.

**Comment by davidrapan** [comment]
It could be due to many reasons:
1. Wrong function used
2. Used wrong register
3. Trying to set the register to incorrect value
4. Just bad luck (No kidding, I already heard from many people that it worked for them after they tried it for x time)

**Reply by PaulOckenden** [reply]
Thanks. I didn’t know whether the “132 bytes” thing would be a clue.
